pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignmentUtils.java

```diff
@@ -439,6 +440,29 @@ public Map<String, Map<String, String>> getNonTierSegmentAssignment() {
     }
   }
 
+  /**
+   * Return the partitionId for an OFFLINE or REALTIME table
+   */
+  public static int getPartitionId(String segmentName, String tableName, TableType tableType, HelixManager helixManager,
+      int numPartitions, @Nullable String partitionColumn) {
+    int partitionId;
+    if (numPartitions == 1) {
+      partitionId = 0;
+    } else {
+      // Uniformly spray the segment partitions over the instance partitions. Use the same logic to calculate the
+      // partitionId if partitionColumn or calculated partitionId is null for OFFLINE and REALTIME tables
+      if (tableType == TableType.OFFLINE) {
+        partitionId = partitionColumn == null ? getDefaultPartitionId(segmentName)
```

Comment:
Is this correct? I think you want to always `% numPartitions`

Reply:
the caller does the modulo, and that's what I saw in the code. in some places the modulo is done (for replica group assignment) and in some places the partitionId is used as is, so decided not to add the modulo to the util directly